connector/src/models/Proposal.ts

```diff
@@ -0,0 +1,63 @@
+import {
+  StakeData,
+  ProposalData,
+  IConvictionVotingConnector,
+  SubscriptionHandler,
+} from '../types'
+import StakeHistory from './StakeHistory'
+
+export default class Proposal {
+  #connector: IConvictionVotingConnector
+
+  readonly id: string
+  readonly number: string
+  readonly name: string
+  readonly link?: string
+  readonly creator: string
+  readonly beneficiary?: string
+  readonly requestedAmount?: string
+  readonly executed: boolean
+  readonly totalTokensStaked: string
+  readonly stakes: StakeData[]
+  readonly appAddress: string
+
+  constructor(data: ProposalData, connector: IConvictionVotingConnector) {
+    this.#connector = connector
+
+    this.id = data.id
+    this.number = data.number
+    this.name = data.name
+    this.link = data.link
+    this.creator = data.creator
+    this.beneficiary = data.beneficiary
+    this.requestedAmount = data.requestedAmount
+    this.executed = data.executed
+    this.totalTokensStaked = data.totalTokensStaked
+    this.stakes = data.stakes
+    this.appAddress = data.appAddress
+  }
+
+  async stakesHistory({ first = 1000, skip = 0 } = {}): Promise<
+    StakeHistory[]
+  > {
+    return this.#connector.stakesHistoryByProposal(
+      this.appAddress,
+      this.id,
```

Comment:
Ah here should be `this.number` since the `proposalId` stored on the `stakeHistory` entity is not the subgraph's Proposal entity id but the id of the proposal itself.

https://github.com/1Hive/conviction-voting-app/blob/e2ec30261538377288eccd84906807460a7a7205/connector/subgraph/src/helpers.ts#L140

```suggestion
      this.number,
```

Reply:
Good catch!